sys/linux/auto.txt

```diff
@@ -278,24 +283,24 @@ getgroups$auto(gidsetsize int32, grouplist ptr[inout, int32])
 getgroups32$auto(gidsetsize int32, grouplist ptr[inout, int32])
 getitimer$auto(which int32, value ptr[inout, __kernel_old_itimerval$auto])
 getpeername$auto(fd fd, usockaddr ptr[inout, sockaddr], usockaddr_len ptr[inout, int32])
-getpgid$auto(pid pid)
-getpriority$auto(which int32, who int32)
+getpgid$auto(pid pid) pid
+getpriority$auto(which int32, who pid)
 getrandom$auto(ubuf ptr[inout, string], len intptr, flags int32)
 getresgid$auto(rgidp ptr[inout, int32], egidp ptr[inout, int32], sgidp ptr[inout, int32])
 getresgid32$auto(rgidp ptr[inout, int32], egidp ptr[inout, int32], sgidp ptr[inout, int32])
 getresuid$auto(ruidp ptr[inout, int32], euidp ptr[inout, int32], suidp ptr[inout, int32])
 getresuid32$auto(ruidp ptr[inout, int32], euidp ptr[inout, int32], suidp ptr[inout, int32])
 getrlimit$auto(_resource int32, rlim ptr[inout, rlimit$auto])
 getrusage$auto(who int32, ru ptr[inout, rusage$auto])
-getsid$auto(pid pid)
+getsid$auto(pid pid) pid
 getsockname$auto(fd fd, usockaddr ptr[inout, sockaddr], usockaddr_len ptr[inout, int32])
-getsockopt$auto(fd fd, level int32, optname int32, optval ptr[inout, string], optlen ptr[inout, int32])
+getsockopt$auto(fd fd, level int32, optname int32, optval ptr[inout, string], optlen ptr[inout, int32]) fd
```

Comment:
Here are all inference explanations:
https://gist.githubusercontent.com/dvyukov/682d525929412fe2df6455cc6b6ff9f0/raw/358d184ce33411863bada8dbe58c2b7a777d4bb7/gistfile1.txt

In particular:
```
  fd <- __do_sys_getsockopt:ret <- __sys_getsockopt:ret <- do_sock_getsockopt:ret <- do_sock_getsockopt:loc.err <- sk_getsockopt:ret <- sk_getsockopt:loc.pidfd <- pidfd_prepare:ret <- __pidfd_prepare:ret <- __pidfd_prepare:loc.pidfd <- get_unused_fd_flags:ret <- __get_unused_fd_flags:ret <- alloc_fd:ret
```

So it's returned here:
https://elixir.bootlin.com/linux/v6.13-rc2/source/net/core/sock.c#L1870

But it's actually an error.

This happened for socketpair as well.

I will add a TODO for `if (x < 0) return x;`.

Reply:
Added a TODO for such cases:
```
+// - Detect cases where returned value is actually an error rather than a resource.
+//   For example, these cases lead to false inference of fd type for returned value:
+//   https://elixir.bootlin.com/linux/v6.13-rc2/source/net/core/sock.c#L1870
+//   https://elixir.bootlin.com/linux/v6.13-rc2/source/net/socket.c#L1742
```